setup.py

```diff
@@ -164,24 +168,33 @@
 
 def check_linker_need_libatomic():
     """Test if linker on system needs libatomic."""
-    code_test = (b'#include <atomic>\n' +
-                 b'int main() { return std::atomic<int64_t>{}; }')
-    cpp_test = subprocess.Popen(['c++', '-x', 'c++', '-std=c++11', '-'],
-                                stdin=PIPE,
-                                stdout=PIPE,
-                                stderr=PIPE)
-    cpp_test.communicate(input=code_test)
-    if cpp_test.returncode == 0:
-        return False
-    # Double-check to see if -latomic actually can solve the problem.
-    # https://github.com/grpc/grpc/issues/22491
-    cpp_test = subprocess.Popen(
-        ['c++', '-x', 'c++', '-std=c++11', '-latomic', '-'],
-        stdin=PIPE,
-        stdout=PIPE,
-        stderr=PIPE)
-    cpp_test.communicate(input=code_test)
-    return cpp_test.returncode == 0
+    compiler = ccompiler.new_compiler()
+    distutils.sysconfig.customize_compiler(compiler)
```

Comment:
Quoting the docs on this function: "The following function is only intended for use within the distutils package."

It sounds like this is meant to be an internal API to `distutils`. It's surprising that it's not marked with an underscore. Does using this open us up to potential breakage as `distutils` does internal refactors?

Reply:
From my code reading, `customize_compiler` is the most important function here to choose compilers from environment variables such as `CC` or `CXX`, which is inevitable to support cross-platform toolchains. How about implementing the code fetching `CXX` variable to find the right compiler instead?

Reply:
Like simply, `cxx = os.environ['CXX'] if 'CXX' in os.environ else 'c++'`

Reply:
That seems reasonable to me. I just want to avoid a situation where we're broken by an update in `distutils`.

Reply:
Using `CXX` works fine for me. The only reason I used `distutils.ccompiler` is I thought it might handle some other edge cases. I'll update the PR.